### PR TITLE
[Hackday][2.7] Added deprecated notes to '_method' and '_scheme' route requirements

### PR DESCRIPTION
**Pull request by dcsg**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | no |
| Fixed tickets | [#12641, #12640] |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Routing/Loader/XmlFileLoader.php [hunk 1]
```diff
@@ -131,6 +131,13 @@ protected function parseRoute(RouteCollection $collection, \DOMElement $node, $p
 
         list($defaults, $requirements, $options, $condition) = $this->parseConfigs($node, $path);
 
+        if (isset($requirements['_method']) || isset($requirements['_scheme'])) {
+            trigger_error(
+                'The router \'_method\' and \'_scheme\' requirements was removed. You should use the new \'methods\' and \'schemes\' settings instead.',
```
Comment (stof) [inline src/Symfony/Component/Routing/Loader/XmlFileLoader.php]: the requirements are not removed, they are deprecated

File: src/Symfony/Component/Routing/Route.php [hunk 2]
```diff
@@ -86,6 +86,13 @@ class Route implements \Serializable
      */
     public function __construct($path, array $defaults = array(), array $requirements = array(), array $options = array(), $host = '', $schemes = array(), $methods = array(), $condition = '')
     {
+        if (isset($requirements['_method']) || isset($requirements['_scheme'])) {
```
Comment (stof) [inline src/Symfony/Component/Routing/Route.php]: this should be done in ``setRequirements`` actually (given that the constructor calls it)

**Comment by stof** [comment]
A few things to change here IMO:

- the XML and Yaml file loaders should convert these requirements to the new format before setting them in the Route, to avoid triggering a second deprecation warning in the Route object.
- tests which are ensuring that the BC layer works should be moved to Legacy* test cases ignoring the deprecation warning by using ``$this->iniSet('error_reporting', -1 & ~E_USER_DEPRECATED)`` in the setUp method
- tests which are ensuring the behavior of the scheme and method requirements themselves should use the new API for them to avoid triggering deprecation warnings.

@dcsg can you update your PR ?

**Comment by dcsg** [comment]
@stof I am a little busy this week, but I can try update it on weekend. Is that ok?

**Comment by stof** [comment]
yes it is

**Comment by Tobion** [comment]
@dcsg I'm finishing this PR.

**Comment by Tobion** [comment]
Closing in favor of #13361 which adds the triggers at different places.

**Comment by dcsg** [comment]
Ok thanks and sorry for the delay

**Comment by Tobion** [comment]
Thanks for your initiative.

**Pull request closed without merging**